### PR TITLE
refactor ALL THE THINGS

### LINKED ISSUE #23: refactor/modularize
in prep for publishing this and opening to contributors, let's break the src and tests up into modules

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #23

### CODE REVIEW

File: src/git.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+use git2::{build::CheckoutBuilder, build::RepoBuilder, Repository as GitRepository,
+           RepositoryInitOptions};
+use quicli::prelude::*;
+use std::fs::remove_dir_all;
```
Comment (killercup) [inline src/git.rs]: wait why are we no longer using the remove_dir_all crate?

File: src/git.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+use std::path::PathBuf;
+use Cli;
+
+pub fn create(project_dir: &PathBuf, args: Cli) -> Result<GitRepository> {
```
Comment (killercup) [inline src/git.rs]: why not take `&Path` and `&Cli`?

Comment (killercup) [reply src/git.rs]: also I'd call this `create_from_git`

File: src/progressbar.rs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+use indicatif::ProgressBar;
```
Comment (killercup) [inline src/progressbar.rs]: how do you feel about `extern crate`s that are not in the root module? I would probably put the `extern crate indicatif;` here

File: src/main.rs [hunk 4]
```diff
@@ -62,131 +60,18 @@ main!(|args: Cli| {
...
+    git::create(&project_dir, args)?;
+    git::remove_history(&project_dir)?;
 
-    for entry in WalkDir::new(&project_dir) {
-        let entry = entry?;
-        if entry.metadata()?.is_dir() {
-            continue;
-        }
+    let mut template = template::new();
+    template = template::substitute(&name, template)?;
```
Comment (killercup) [inline src/main.rs]: For what is currently happening here the `template::new()` is confusing -- it is a (constant) constructor of an empty object that is immediately passed to `substitute`. And `substitute` is only ever called with this (still constant) value.

I'd change this whole thing to `let template = template::substitute(&name);`

File: src/git.rs (hunk 1 above)
Comment (ashleygwilliams) [reply src/git.rs]: oh crap. mistakes were made. thanks for catching!

File: src/git.rs (hunk 2 above)
Comment (ashleygwilliams) [reply src/git.rs]: re Cli, i just didn't need to. but don't really care. the module is called git, so it reads `git::create` - i'd rather not have the redundancy

File: src/progressbar.rs (hunk 3 above)
Comment (ashleygwilliams) [reply src/progressbar.rs]: not generally a huge fan :/

File: src/main.rs (hunk 4 above)
Comment (ashleygwilliams) [reply src/main.rs]: wfm

**Pull request merged**